crates/shared/src/price_estimation/trade_verifier.rs

```diff
@@ -287,17 +287,60 @@ impl TradeVerifier {
     /// trade.
     async fn prepare_state_overrides(
         &self,
-        verification: &Verification,
+        verification: &mut Verification,
         query: &PriceQuery,
         trade: &TradeKind,
     ) -> Result<HashMap<H160, StateOverride>> {
+        let mut overrides = HashMap::default();
```

Comment:
As the items count of the hashmap is known we could use `with_capacity(6)` here (this will avoid 1 memory reallocation).